core-java/src/test/java/com/baeldung/arrays/ArraysTest.java

```diff
@@ -0,0 +1,148 @@
+                .allMatch(el -> "once".equals(el)));
+    }
+
+    @Test
+    public void testArraysEquals(){
+        Assert.assertTrue(Arrays.equals(new String[]{"once","upon","a","time"}, intro));
+        Assert.assertFalse(Arrays.equals(new String[]{"once","upon","a", null}, intro));
+    }
+
+    @Test
+    public void testArraysDeepEquals() {
+        String[] end = {"the", "end"};
+        Object[] story =
+                new Object[] { intro, new String[] { "chapter one", "chapter two" }, end };
+        Object[] copy =
+                new Object[] { intro, new String[] { "chapter one", "chapter two" }, end };
+
+        Assert.assertTrue(Arrays.deepEquals(story, copy));
+        Assert.assertFalse(Arrays.equals(story, copy));
+    }
+
+    @Test
+    public void testArraysSort(){
+        String[] sorted = Arrays.copyOf(intro, 4);
+        Arrays.sort(sorted);
+
+        Assert.assertArrayEquals(new String[]{"a", "once", "time", "upon"}, sorted);
+    }
+
+    @Test
+    public void testArraysBinarySearch(){
+        String[] sorted = Arrays.copyOf(intro, 4);
+        Arrays.sort(sorted);
+        int exact = Arrays.binarySearch(sorted, "time");
+        int caseInsensitive = Arrays.binarySearch(sorted, "TiMe", String::compareToIgnoreCase);
+
+        Assert.assertEquals("time", sorted[exact]);
+        Assert.assertEquals(2, exact);
+        Assert.assertEquals(exact, caseInsensitive);
+    }
+
+    @Test
+    public void testArraysHashCode(){
+        int beforeChange = Arrays.hashCode(intro);
+        intro[3] = null;
+        int afterChange = Arrays.hashCode(intro);
+
+        Assert.assertNotEquals(beforeChange, afterChange);
+    }
+
+    @Test
+    public void testArraysDeepHash(){
+        Object[] looping = new Object[]{ intro, intro };
+        int deepHashBefore = Arrays.deepHashCode(looping);
+        int hashBefore = Arrays.hashCode(looping);
+
+        intro[3] = null;
+
+        int hashAfter = Arrays.hashCode(looping);
+        int deepHashAfter = Arrays.deepHashCode(looping);
+
+        Assert.assertEquals(hashAfter, hashBefore);
+        Assert.assertNotEquals(deepHashAfter, deepHashBefore);
+    }
+
+    @Test
+    public void testStream(){
+        Assert.assertEquals(Arrays.stream(intro).count(), 4);
+
+        exception.expect(ArrayIndexOutOfBoundsException.class);
+        Arrays.stream(intro, 2, 1).count();
+    }
+
+    @Test
+    public void testSetAll(){
+        String[] longAgo = new String[4];
+        Arrays.setAll(longAgo, i -> this.getWord(i));
+        Assert.assertArrayEquals(longAgo, new String[]{"A","long","time","ago"});
+    }
+
+    public String getWord(int i){
+        String[] introWords = new String[]{"A","long","time","ago"};
+        return introWords[i];
+    }
+
+    @Test
+    public void testArraysToString(){
+        Assert.assertEquals("[once, upon, a, time]", Arrays.toString(intro));
+    }
+
+    @Test
+    public void testDeepToString(){
+        String[] end = {"the", "end"};
```

Comment:
Will you take a minute to go through and make sure your spacing is consistent? It will make your (already very clean code) easier to follow.

For example, on line 133, you do {"the", "end"} and on line 135, you put some more spacing around each element. I personally like it with the extra spacing.